stonesoup/initiator/simple.py

```diff
@@ -369,16 +369,17 @@ def initiate(self, detections, timestamp, **kwargs):
         tracks = self.initiator.initiate(detections, timestamp, **kwargs)
 
         for track in tracks:
-            mixture = [
-                TaggedWeightedGaussianState(
-                    state_vector=track.state_vector,
-                    covar=track.covar,
-                    weight=Probability(1),
-                    timestamp=track.timestamp,
-                    tag=[])]
-            track[-1] = GaussianMixtureUpdate(
-                hypothesis=track.hypothesis,
-                components=mixture)
+            for n, state in enumerate(track):
+                mixture = [
+                    TaggedWeightedGaussianState(
+                        state_vector=state.state_vector,
+                        covar=state.covar,
+                        weight=Probability(1),
+                        timestamp=state.timestamp,
+                        tag=[])]
+                track[n] = GaussianMixtureUpdate(
+                    hypothesis=track.hypothesis,
```

Comment:
Maybe this should get per state as well?
```suggestion
                    hypothesis=getattr(state, 'hypothesis', None),
```

Reply:
Good point, changed.